### PR TITLE
Merge vonmises into stats pyx

### PR DESCRIPTION
**Pull request by anntzer**
Suggested in #5936, only makes sense if that PR is merged first.

### CODE REVIEW

**Comment by rgommers** [comment]
One test failure:

    ======================================================================
    FAIL: test_distributions.test_all_distributions('vonmises', (101,), 0.01)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/home/travis/virtualenv/python3.5.0/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
        self.test(*self.arg)
      File "/home/travis/virtualenv/python3.5.0/lib/python3.5/site-packages/scipy/stats/tests/test_distributions.py", line 65, in check_distribution
        "; alpha = " + str(alpha) + "\nargs = " + str(args))
      File "/home/travis/virtualenv/python3.5.0/lib/python3.5/site-packages/numpy/testing/utils.py", line 71, in assert_
        raise AssertionError(smsg)
    AssertionError: D = 0.271825589958; pval = 0.0; alpha = 0.01
    args = (101,)

**Comment by rgommers** [comment]
merging the files make sense to me

**Comment by anntzer** [comment]
Should be fixed now.  Also rebased on the new #5936.

**Comment by Eric89GXL** [comment]
Travis is unhappy about some minor doc issues:

https://travis-ci.org/scipy/scipy/jobs/116304847#L3460

**Comment by anntzer** [comment]
fixed

File: scipy/stats/_stats.pyx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    result = np.empty_like(bx, float)
+
+    c_small_k = bk < CK
+    temp = result[c_small_k]
+    temp_xs = bx[c_small_k].astype(float)
+    temp_ks = bk[c_small_k].astype(float)
+    for i in range(len(temp)):
+        p = <int>(1 + a1 + a2 * temp_ks[i] - a3 / (temp_ks[i] + a4))
+        temp[i] = von_mises_cdf_series(temp_ks[i], temp_xs[i], p)
+        temp[i] = 0 if temp[i] < 0 else 1 if temp[i] > 1 else temp[i]
```
Comment (ev-br) [inline scipy/stats/_stats.pyx]: Whoa. Can't believe this is legal. I had to read it out loud, then again, and then consult the red-colored part of the diff. This is very cool, but please have some mercy for us mere mortals.

File: scipy/stats/_stats.pyx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        sn, cn = sn * c - cn * s, cn * c + sn * s
+        R = 1. / (2 * n / k + R)
+        V = R * (sn / n + V)
+
+    with cython.cdivision(True):
+        return 0.5 + x / (2 * PI) + V / PI
+
+
+cdef von_mises_cdf_normalapprox(k, x):
+    b = math.sqrt(2 / PI) / scipy.special.i0e(k) # Check for negative k
```
Comment (ev-br) [inline scipy/stats/_stats.pyx]: Again, it's probably just me being thick. What is this comment about, what's the deal with k < 0?

Comment (anntzer) [inline scipy/stats/_stats.pyx]: This is copypasted from the original file, I have no idea.

File: scipy/stats/_stats.pyx (hunk 1 above)
Comment (anntzer) [inline scipy/stats/_stats.pyx]: I thought this actually reads quite "intuitively" (as if it was English)?

Comment (ev-br) [inline scipy/stats/_stats.pyx]: Well, you can probably attribute my confusion to ESL.

Comment (anntzer) [inline scipy/stats/_stats.pyx]: So, do you want me to change this?  I think it's more readable than `min(max(temp[i], 0), 1)`...

Comment (ev-br) [inline scipy/stats/_stats.pyx]: If it were my PR, I would just write out two nested if-then-else clauses. Which is not cute etc. But this is your PR, and I won't hold it for this.

File: scipy/stats/_stats.pyx (hunk 2 above)
Comment (ev-br) [inline scipy/stats/_stats.pyx]: As a guess, k>0 is enforced by `argcheck`, and this routine just never sees negative k. If this is the case, the comment can be removed. (Not necessary for this PR)

File: scipy/stats/_stats.pyx (hunk 1 above)
Comment (anntzer) [inline scipy/stats/_stats.pyx]: I'll just leave it as it is then, unless someone else objects.

Comment (Eric89GXL) [inline scipy/stats/_stats.pyx]: np.clip is pretty clear

Comment (anntzer) [inline scipy/stats/_stats.pyx]: Although this is unlikely to be a bottleneck, it still looks like gratuitiously invoking the Python runtime...

Comment (Eric89GXL) [inline scipy/stats/_stats.pyx]: Ahh right Cython, never mind. Personally I find the max min call more
readable but this is fine by me.

**Comment by pv** [comment]
Needs rebase, the other PR was merged.

**Comment by anntzer** [comment]
Done.

**Comment by ev-br** [comment]
Ok, merging. Thanks @anntzer

**Pull request merged**